src/renderer/atlas/BackendD3D.cpp

```diff
@@ -2066,6 +2067,38 @@ size_t BackendD3D::_drawCursorForegroundSlowPath(const CursorRect& c, size_t off
     return addedInstances;
 }
 
+void BackendD3D::_drawSearchSelections(const RenderingPayload& p)
+{
+    u16 y = 0;
+
+    for (const auto& row : p.rows)
+    {
+        if (!row->searchSelections.empty())
+        {
+            for (auto s : row->searchSelections)
+            {
+                if (s.from != row->selectionFrom)

```

Comment:
I was running into an issue where the rectangle would get double colored (I think it ended up in a blend of the selection and the highlight color).  It would get the normal selection coloring from _drawSelection and then it would get higlight coloring from _drawSelections.

(s.from != row->selectionFrom) was a bit weird,  it was an attempt to still color any other highlights on the line that wasn't the main selection.

Reply:
It looks like this had a bug in it where if the search result started on the first character of the line it would not get highlighted since selection from would be 0 for all rows.

https://github.com/microsoft/terminal/pull/16227/commits/d8df3d90045f6ad153efc9a07878908bd0aaa4f9